tuf/signed/verify.go

```diff
@@ -105,3 +106,31 @@ func VerifySignature(msg []byte, sig data.Signature, pk data.PublicKey) error {
 	}
 	return nil
 }
+
+// VerifyPublicKeyMatchesPrivateKey checks that the specified private key and public key together form a valid keypair.
+func VerifyPublicKeyMatchesPrivateKey(privKey data.PrivateKey, pubKey data.PublicKey) error {
```

Comment:
@dnwake I don't have one on-hand -- are you getting an empty public key back?  What did you try using?  I'll try to come up with a sample too